warp/_src/context.py

```
@@ -4927,10 +4927,12 @@ def __init__(self, runtime, alias, ordinal=-1, is_primary=False, context=None):
             else:
                 self.is_ipc_supported = False
             if warp.config.enable_mempools_at_init:
-                # enable if supported
-                self.is_mempool_enabled = self.is_mempool_supported
+                # HIP/ROCm: auto-enabling the memory pool conflicts with PyTorch's
+                # ROCm allocator when used in the same process. Apply opt-in only
+                # for HIP devices; CUDA mempool behavior is completely unchanged.
+                hip_pool_ok = not self.is_hip or _os.environ.get(WARP_HIP_MEMPOOL_ENABLE, 0) == 1
```

Ruff (0.15.21)

[error] Undefined name `_os`

(F821)